benchmark_utils/utils.py

```diff
@@ -103,32 +103,55 @@ def __getitem__(self, idx):
         return image, label
 
 
-def get_model_and_batch_size(dataset_name, n_classes):
+def get_deep_model(dataset_name, n_classes):
     """
-    Get the appropriate model and batch size for a given dataset.
+    Returns the appropriate deep learning model for the specified dataset.
 
     Args:
-        dataset_name (str): Name of the dataset ('mnist_usps', 'office31', 'officehome', or 'bci').
-        n_classes (int): Number of classes in the dataset.
+        dataset_name (str): Name of the dataset. Must be one of:
+            'mnist_usps', 'office31', 'officehome', or 'bci'.
+            n_classes (int): Number of output classes for the model.
 
     Returns:
-        tuple: A tuple containing:
-            - model: The appropriate neural network model for the dataset.
-            - batch_size (int): The recommended batch size for the dataset.
+        dict: A dictionary containing the following keys:
+            - 'module': The neural network module suited for the dataset.
+            - 'batch_size' (int): The recommended batch size for the dataset.
+            - 'max_epochs' (int): The maximum number of training epochs.
+            - 'lr' (float): The recommended learning rate.
 
     Raises:
         ValueError: If an unsupported dataset name is provided.
     """
 
     dataset_configs = {
-        'mnist_usps': {'batch_size': 256, 'model': ShallowConvNet(n_classes=n_classes)},
-        'office31': {'batch_size': 128, 'model': ResNet18Net(n_classes=n_classes)},
-        'officehome': {'batch_size': 128, 'model': ResNet18Net(n_classes=n_classes)},
-        'bci': {'batch_size': 256, 'model': FBCSPNet(n_chans=22, n_classes=n_classes, input_window_samples=1125)}
+        'mnist_usps': {
+            'batch_size': 256,
+            'module': ShallowConvNet(n_classes=n_classes),
+            'max_epochs': 14,
+            'lr': 1
```

Comment:
We should also add in the dataset_configs the LR scheduler since for BCI we'll need a cosineAnealing instead of a StepLR

Reply:
Ok, so maybe we merge this PR first and then add LR scheduler in the BCI PR?

Reply:
works with me